pybliometrics/scopus/serial_search.py

```diff
@@ -1,49 +1,44 @@
-from collections import namedtuple, OrderedDict
+from collections import OrderedDict
 
 from pybliometrics.scopus.superclasses import Search
 
 
 class SerialSearch(Search):
     @property
     def results(self):
-        """A list of namedtuples representing results of serial search. The
-        number of fields may vary from one search query to another depending
-        on the length of yearly data Note: can be empty.
+        """A list of OrderedDicts representing results of serial search. The
+        number of keys may vary from one search result to another depending
+        on the length of yearly data. Note: can be empty.
         """
         out = []
-        search_results = self._json['serial-metadata-response'].get('entry')
-        if search_results:
-            # Namedtuple fields from union of keys of each dictionary in result list
-            fields = [_clean_field_name(i) for i in _form_fields(search_results)]
-            serial = namedtuple('Serial', fields)
-            for i in search_results:
-                # OrderedDict to populate with individual serial data
-                obs = OrderedDict.fromkeys(fields)
-                for key in i:
-                    if key != '@_fa' and i[key]:
-                        if key == 'subject-area':
-                            subject_data = _merge_subject_data(i[key])
-                            obs['subject_area_codes'] = subject_data[0]
-                            obs['subject_area_abbrevs'] = subject_data[1]
-                            obs['subject_area_names'] = subject_data[2]
-                        elif key == 'SNIPList' or key == 'SJRList':
-                            for j in _retrieve_source_rankings(i[key]):
+        search_results = self._json['serial-metadata-response'].get('entry', [])
+        for i in search_results:
+            # OrderedDict to populate with individual serial data
+            obs = OrderedDict()
+            for key in i:
```

Comment:
If `i` is a dict, we should do three things:
1. in line 14, the default value should be `{}`, not `[]`
2. in line 18, we should iterate over `i` using both key _and_ value, so to not use `i[key]` in the loop
3. since `i` is rather reserved for indices, we should rename it to make clear it's an element in a list, e.g. `item` (I think that's what other classes are using)

Reply:
1. I do not see why default value should be `{}` and not `[]`. The `search_results` is list of dictionaries.

Reply:
Oh yes, you're absolutely right!